<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="LightpadTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTipDuration">
       <number>0</number>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>-1</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideLeft</enum>
      </property>
      <property name="usesScrollButtons">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="backgroundBottom" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>36</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>8</number>
       </property>
       <property name="leftMargin">
        <number>12</number>
       </property>
       <property name="rightMargin">
        <number>12</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
        <item>
         <widget class="QToolButton" name="runButton">
          <property name="styleSheet">
           <string notr="true">QToolButton {
qproperty-icon: url(":/resources/icons/run.png");
}</string>
          </property>
          <property name="toolTip">
           <string>Run</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="debugButton">
          <property name="styleSheet">
           <string notr="true">QToolButton {
qproperty-icon: url(":/resources/icons/debug.png");
}</string>
          </property>
          <property name="toolTip">
           <string>Debug</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       <item>
        <widget class="QToolButton" name="magicButton">
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>28</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QToolButton {
qproperty-icon: url(":/resources/icons/wand_dark.png");
}</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>5</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QToolButton" name="languageHighlight">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Choose lang.</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>5</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QToolButton" name="tabWidth">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Tab Width: 4</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>5</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="rowCol">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Ln 0, Col 0</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_Window"/>
    <addaction name="actionNew_File"/>
    <addaction name="actionOpen_File"/>
    <addaction name="actionOpen_Project"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
    <addaction name="actionSave_All"/>
    <addaction name="separator"/>
    <addaction name="actionClose_Tab"/>
    <addaction name="actionClose_All_Tabs"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuTransform_Text">
     <property name="title">
      <string>Transform Text</string>
     </property>
     <addaction name="actionTransform_Uppercase"/>
     <addaction name="actionTransform_Lowercase"/>
     <addaction name="actionTransform_Title_Case"/>
    </widget>
    <widget class="QMenu" name="menuSort_Lines">
     <property name="title">
      <string>Sort Lines</string>
     </property>
     <addaction name="actionSort_Lines_Ascending"/>
     <addaction name="actionSort_Lines_Descending"/>
    </widget>
    <addaction name="actionToggle_Undo"/>
    <addaction name="actionToggle_Redo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="menuTransform_Text"/>
    <addaction name="menuSort_Lines"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuFolding">
     <property name="title">
      <string>Folding</string>
     </property>
     <addaction name="actionFold_Current"/>
     <addaction name="actionUnfold_Current"/>
     <addaction name="separator"/>
     <addaction name="actionFold_All"/>
     <addaction name="actionUnfold_All"/>
     <addaction name="separator"/>
     <addaction name="actionFold_Comments"/>
     <addaction name="actionUnfold_Comments"/>
    </widget>
    <addaction name="actionToggle_Full_Screen"/>
    <addaction name="actionToggle_Minimap"/>
    <addaction name="actionToggle_Terminal"/>
    <addaction name="actionToggle_Source_Control"/>
    <addaction name="actionToggle_Word_Wrap"/>
    <addaction name="actionToggle_Vim_Mode"/>
    <addaction name="separator"/>
    <addaction name="menuFolding"/>
    <addaction name="separator"/>
    <addaction name="actionSplit_Horizontally"/>
    <addaction name="actionSplit_Vertically"/>
    <addaction name="actionClose_Editor_Group"/>
    <addaction name="actionFocus_Next_Group"/>
    <addaction name="actionFocus_Previous_Group"/>
    <addaction name="actionUnsplit_All"/>
    <addaction name="separator"/>
    <addaction name="actionIncrease_Font_Size"/>
    <addaction name="actionDecrease_Font_Size"/>
    <addaction name="actionReset_Font_Size"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_Qt"/>
    <addaction name="actionKeyboard_shortcuts"/>
   </widget>
   <widget class="QMenu" name="menuSearch">
    <property name="title">
     <string>Search</string>
    </property>
    <addaction name="actionFind_in_file"/>
    <addaction name="actionFind_in_project"/>
    <addaction name="separator"/>
    <addaction name="actionReplace_in_file"/>
    <addaction name="actionReplace_in_project"/>
    <addaction name="separator"/>
    <addaction name="actionGo_to_Line"/>
   </widget>
    <widget class="QMenu" name="menuRun">
     <property name="title">
      <string>Run</string>
     </property>
     <addaction name="actionRun_file_name"/>
     <addaction name="actionEdit_Configurations"/>
    </widget>
    <widget class="QMenu" name="menuDebug">
     <property name="title">
      <string>Debug</string>
     </property>
     <addaction name="actionDebug_file_name"/>
     <addaction name="actionEdit_Debug_Configurations"/>
    </widget>
   <widget class="QMenu" name="menuFormat">
    <property name="title">
     <string>Format</string>
    </property>
    <addaction name="actionFormat_Document"/>
    <addaction name="actionEdit_Format_Configurations"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuSearch"/>
   <addaction name="menuRun"/>
    <addaction name="menuDebug"/>
    <addaction name="menuFormat"/>
    <addaction name="menuHelp"/>
   </widget>
  <action name="actionToggle_Undo">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionToggle_Redo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionToggle_Full_Screen">
   <property name="text">
    <string>Toggle Full Screen</string>
   </property>
  </action>
  <action name="actionIncrease_Font_Size">
   <property name="text">
    <string>Increase Font Size</string>
   </property>
  </action>
  <action name="actionDecrease_Font_Size">
   <property name="text">
    <string>Decrease Font Size</string>
   </property>
  </action>
  <action name="actionReset_Font_Size">
   <property name="text">
    <string>Reset Font Size</string>
   </property>
  </action>
  <action name="actionNew_Window">
   <property name="text">
    <string>New Window</string>
   </property>
  </action>
  <action name="actionNew_File">
   <property name="text">
    <string>New File</string>
   </property>
  </action>
  <action name="actionOpen_File">
   <property name="text">
    <string>Open File</string>
   </property>
  </action>
  <action name="actionOpen_Project">
   <property name="text">
    <string>Open Project</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cut</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste</string>
   </property>
  </action>
  <action name="actionSave_All">
   <property name="text">
    <string>Save All</string>
   </property>
  </action>
  <action name="actionClose_Tab">
   <property name="text">
    <string>Close Tab</string>
   </property>
  </action>
  <action name="actionClose_All_Tabs">
   <property name="text">
    <string>Close All Tabs</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About Lightpad</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionFind_in_file">
   <property name="text">
    <string>Find in file</string>
   </property>
  </action>
  <action name="actionFind_in_project">
   <property name="text">
    <string>Find in project</string>
   </property>
  </action>
  <action name="actionReplace_in_file">
   <property name="text">
    <string>Replace in file</string>
   </property>
  </action>
  <action name="actionReplace_in_project">
   <property name="text">
    <string>Replace in project</string>
   </property>
  </action>
  <action name="actionKeyboard_shortcuts">
   <property name="text">
    <string>Keyboard shortcuts</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences</string>
   </property>
  </action>
  <action name="actionRun_file_name">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/resources/icons/run.png</normaloff>:/resources/icons/run.png</iconset>
   </property>
   <property name="text">
    <string>Run file_name</string>
   </property>
  </action>
  <action name="actionDebug_file_name">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/resources/icons/debug.png</normaloff>:/resources/icons/debug.png</iconset>
   </property>
   <property name="text">
    <string>Debug file_name</string>
   </property>
  </action>
  <action name="actionEdit_Configurations">
   <property name="text">
    <string>Edit Configurations</string>
   </property>
  </action>
  <action name="actionEdit_Debug_Configurations">
   <property name="text">
    <string>Edit Debug Configurations</string>
   </property>
  </action>
  <action name="actionFormat_Document">
   <property name="text">
    <string>Format Document</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+F</string>
   </property>
  </action>
  <action name="actionEdit_Format_Configurations">
   <property name="text">
    <string>Edit Format Configurations</string>
   </property>
  </action>
  <action name="actionGo_to_Line">
   <property name="text">
    <string>Go to Line...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionToggle_Minimap">
   <property name="text">
    <string>Toggle Minimap</string>
   </property>
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionSplit_Horizontally">
   <property name="text">
    <string>Split Editor Right</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+\</string>
   </property>
  </action>
  <action name="actionSplit_Vertically">
   <property name="text">
    <string>Split Editor Down</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, Ctrl+\</string>
   </property>
  </action>
  <action name="actionClose_Editor_Group">
   <property name="text">
    <string>Close Editor Group</string>
   </property>
  </action>
  <action name="actionFocus_Next_Group">
   <property name="text">
    <string>Focus Next Editor Group</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, Ctrl+Right</string>
   </property>
  </action>
  <action name="actionFocus_Previous_Group">
   <property name="text">
    <string>Focus Previous Editor Group</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, Ctrl+Left</string>
   </property>
  </action>
  <action name="actionUnsplit_All">
   <property name="text">
    <string>Reset Editor Groups</string>
   </property>
  </action>
  <action name="actionToggle_Terminal">
   <property name="text">
    <string>Toggle Terminal</string>
   </property>
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="shortcut">
    <string>Ctrl+`</string>
   </property>
  </action>
  <action name="actionToggle_Source_Control">
   <property name="text">
    <string>Toggle Source Control</string>
   </property>
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+G</string>
   </property>
  </action>
  <action name="actionTransform_Uppercase">
   <property name="text">
    <string>UPPERCASE</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+U</string>
   </property>
  </action>
  <action name="actionTransform_Lowercase">
   <property name="text">
    <string>lowercase</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="actionTransform_Title_Case">
   <property name="text">
    <string>Title Case</string>
   </property>
  </action>
  <action name="actionSort_Lines_Ascending">
   <property name="text">
    <string>Sort Ascending</string>
   </property>
  </action>
  <action name="actionSort_Lines_Descending">
   <property name="text">
    <string>Sort Descending</string>
   </property>
  </action>
  <action name="actionToggle_Word_Wrap">
   <property name="text">
    <string>Toggle Word Wrap</string>
   </property>
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="shortcut">
    <string>Alt+Z</string>
   </property>
  </action>
  <action name="actionToggle_Vim_Mode">
   <property name="text">
    <string>Toggle Vim Mode</string>
   </property>
   <property name="checkable">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionFold_Current">
   <property name="text">
    <string>Fold Current Block</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+[</string>
   </property>
  </action>
  <action name="actionUnfold_Current">
   <property name="text">
    <string>Unfold Current Block</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+]</string>
   </property>
  </action>
  <action name="actionFold_All">
   <property name="text">
    <string>Fold All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, Ctrl+0</string>
   </property>
  </action>
  <action name="actionUnfold_All">
   <property name="text">
    <string>Unfold All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, Ctrl+J</string>
   </property>
  </action>
  <action name="actionFold_Comments">
   <property name="text">
    <string>Fold All Comments</string>
   </property>
  </action>
  <action name="actionUnfold_Comments">
   <property name="text">
    <string>Unfold All Comments</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LightpadTabWidget</class>
   <extends>QTabWidget</extends>
   <header>core/lightpadtabwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections/>
</ui>
